### ISSUE TITLE
Tracker crash when a client refreshes frontend video page

### ISSUE THREAD

**Issue by ckcr4lyf** [Novage/wt-tracker#3]
Hey, first of all thanks for a great repository. 

I noticed what I think is a bug, when I was testing the tracker. I was making use of a minimal frontend demo from p2p-media-loader, and wanted to try my own tracker.

When I use wt-tracker, I was testing on a remote server, with one instance on my PC and one on my mobile. PC is on Lan, and Mobile is on Wifi (different networks, but more on this later).

So initially when the tracker is first loaded, communication works great, I can see load being shared, and p2p requests / announces being sent. But if I reload the tab / website in either my PC or my phone, an exception is thrown which I have traced back to https://github.com/Novage/wt-tracker/blob/0903e41525a2f6fd27a42f22633dec898140cbfb/lib/uws-tracker.ts#L127

The error specifically is (in the tracker console): 
```
Invalid access of closed uWS.WebSocket/SSLWebSocket.
```

And p2p communication stops. When I reloaded on my phone, (so I could check PC chrome console), I saw an Ice Connection error as well (might be related to old peer data which doesn't exist? And since tracker crashed cannot get new peer, so unloaded PC tries to communicate with old peer)

```
2p-media-loader-core.min.js:1 p2pml:media-peer peer error +888ms 06ddf104bb2ff52210bc9c0394b7f48f912a420d Error: Ice connection failed.
    at l (p2p-media-loader-core.min.js:1)
    at c._onIceStateChange (p2p-media-loader-core.min.js:1)
    at RTCPeerConnection.r._pc.oniceconnectionstatechange (p2p-media-loader-core.min.js:1) n.MediaPeer {_events: {…}, _eventsCount: 10, _maxListeners: undefined, peer: c, settings: {…}, …}
p2p-media-loader-core.min.js:1 p2pml:media-peer peer close +1ms 06ddf104bb2ff52210bc9c0394b7f48f912a420d n.MediaPeer {_events: {…}, _eventsCount: 10, _maxListeners: undefined, peer: c, settings: {…}, …}
```

and then rightly discards the non-responsive peer. 

I have tried the same refresh scenario with bitorrent-tracker, and have not observed a crash, so I think it is specific to this library, rather than wss based trackers. 

Further, a simple restart of the tracker and waiting around ~30 secs the two devices connect again and communication begins. 

So, theoretically I guess using pm2 it is possible to restart on this crash, but if it could be avoided then its probably better.

Lastly, I've been reading the code since I am extremely interested in P2P and had initially thought of making something on my own before I discovered your amazing work. However, I'm having some trouble reading it initially, since the work I've done in node.js is mostly server-side, and I'm unfamiliar with `.ts` and some files just containing what seem like definitions. If you could point me in the right direction I'd greatly appreciate it, and hopefully I can contribute a lot of features to the repository!

Thanks

P.S. on my two devices I have two separate public IPs, but in the console I can see the IP that it gets of my phone as a private one (both are connected to different networks but by my University), was wondering how this works? Not asking for a detailed explanation, but just a few keywords I can search and read up on.

**Comment by mrlika** [comment]
`.ts` is TypeScript that is in the top 10 of most popular languages our days. Very similar to JavaScript but strongly typed and requires compilation. If you know modern JavaScript (ES6/ES8) then you know 90% of TypeScript.

Regarding the bug `Invalid access of closed uWS.WebSocket/SSLWebSocket` - I can reproduce it. It appeared after switching to `uWebSockets.js v15.5.0`. Looks like the bug is in the library, not in `wt-tracker` code.

Quick solution - downgrade back to `uWebSockets.js v15.4.0`

**Comment by mrlika** [comment]
There was a typo in my code. Fixed in 296f74d36fd4d46746eaff76828af19c96ca22dc and now it should work correctly.

**Comment by ckcr4lyf** [comment]
Ah, great, thanks. Working perfectly with refreshes now. Will look into typescript and modular programming, and can hopefully make myself useful as well! Cheers
